### PR TITLE
Favorite button

### PR DESCRIPTION
**Pull request by Pinguinflieger**

### CODE REVIEW

File: marketplace_tracker.js [hunk 1]
```diff
@@ -70,6 +71,31 @@ function getSellPrice() {
     }
 }
 
+function favoriteButton(currentItem){
+    if(document.getElementById("marketplace-refresh-button").nextSibling==document.getElementById("marketplace-item-list-searchbar")){
```
Comment (IceFreez3r) [inline marketplace_tracker.js:75]: Reduce unnessecary indentation with
```js
if (.. != ..) {
  return;
}

File: marketplace_tracker.js [hunk 2]
```diff
@@ -184,6 +223,43 @@ function formatNumber(price) {
...
+
+function highlightFavoriteBuy() {
+    getFavoriteList().then(favoritesList => {
+        let items = document.getElementsByClassName('marketplace-content');
+        let item = items[0].firstChild.firstChild;
+        highlightFavorite(item, favoritesList);
+        while (item.nextElementSibling != null) {
+            item = item.nextElementSibling;
+            highlightFavorite(item, favoritesList);
+        }
```
Comment (IceFreez3r) [inline marketplace_tracker.js:259]: Same here

File: marketplace_tracker.js [hunk 3]
```diff
@@ -70,6 +71,31 @@ function getSellPrice() {
...
+            console.log(response);
+            document.getElementById("marketplace-refresh-button").insertAdjacentHTML("afterend", favoriteTemplate(response.isFavorite));
+            let favoriteButton = document.getElementById("marketplace-favorite-button")
+            favoriteButton.addEventListener('click', function(){
+                sendMessage({
+                    type: 'toggle-favorite',
+                    item: currentItem
+                }).then(response => {
+                    console.log(response.success);
+                    console.log(response.isFavorite);
```
Comment (IceFreez3r) [inline marketplace_tracker.js:90]: A lot of leftover `console.log`s in the code.
Btw you can just log response directly here. Less console output

File: marketplace_tracker.js [hunk 4]
```diff
@@ -192,8 +268,10 @@ function scanMarketList() {
         }
         // Don't scan the sell container, this is done by scanSellList()
         if (items[0].firstChild.className == "marketplace-sell-container") {
+            highlightFavoriteSell();
```
Comment (IceFreez3r) [inline marketplace_tracker.js:271]: Add this line to `scanSellList()` instead

**Comment by IceFreez3r** [comment]
Weird highlighting for dungeon keys
grafik

File: marketplace_tracker.js (hunk 4 above)
Comment (Pinguinflieger) [reply marketplace_tracker.js:268]: I think it's more efficient to move `scanSellList()` to this if instead.

File: marketplace_tracker.js [hunk 5]
```diff
@@ -184,6 +223,43 @@ function formatNumber(price) {
...
+
+function highlightFavoriteSell() {
+    getFavoriteList().then(favoritesList => {
+        let items = document.getElementsByClassName('marketplace-sell-items');
+        let item = items[0].firstChild;
+        highlightFavorite(item, favoritesList);
+        while (item.nextElementSibling != null) {
+            item = item.nextElementSibling;
+            highlightFavorite(item, favoritesList);
+        }
```
[the remark this answers is not in the document]
Comment (Pinguinflieger) [reply marketplace_tracker.js:244]: it _feels_ slower, didn't test it tho

**Pull request merged**